### DISCUSSION TITLE
Checkbox/radio indicator loading state

### DISCUSSION THREAD

**Discussion by slovenianGooner** [livewire/flux#884] [Feature Requests]
Hi guys,

we have an interesting use-case for a checkbox to display a loading spinner. Say what? Yes, a loading spinner.

So basically we have a checkbox group with variant cards where we display a checkbox indicator.

<img width="1006" alt="Screenshot 2024-12-18 at 08 56 50" src="https://github.com/user-attachments/assets/b6d6e3e0-745f-4010-bb4a-f771cd5ea7c7" />

When we select the checkbox, we preform some checks in the background - basically a Livewire request is made that determines if there are any conflicts with other "options" and then we display a modal. This sometimes takes up to 500ms and we want to inform our user that something is happening.

We tried to use wire:loading with wire:target to give us a loading indicator until the request is processed. The problem is that the checkbox state ie. checked reacts to the `data-checked` attribute on `ui-checkbox` and immediately shows the checked status, even before wire:loading occurs. Now, this is probably not a problem on really fast computers, but if I slow it down with Chrome we see that it just doesn't look right (note I've used the 20x slowdown here).

https://github.com/user-attachments/assets/3e6e3dd3-d822-43b8-acb7-a9c50408550d

We've tried to fiddle with CSS to see if we can delay this checked status by overriding the Flux CSS to be linked to the `checked` property which is set after the request is done, but we just couldn't make it work. 

This is what we are doing right now, but like said, the delay is there.

```php
<flux:checkbox value="{{ $index }}"
               wire:change="select('{{ $index }}')"
               :checked="$selected">
    <div class="relative">
        <flux:checkbox.indicator wire:target="select('{{ $index }}')"
                                 wire:loading.remove/>
        <div class="absolute top-0 right-0" wire:loading wire:target="select('{{ $index }}')">
            <flux:icon.loading class="size-[1.125rem] text-[var(--color-accent)]"/>
        </div>
    </div>
</flux:checkbox>
```

Ideally, there would be a way to indicate to the "indicator" that is should expect a loading state upon click (or maybe the checkbox too) and then the spinner can show before the checked status does.

Hopefully it is clear what we are facing, if you have any questions feel free to ask.

Thanks for the consideration :)

**Comment by jeffchown** [comment]
@slovenianGooner I believe the `wire:change` is fired *after* the checkbox is updated/changed.  Have you tried changing the `wire:change` to `wire:click` to call the method first?

**Comment by slovenianGooner** [comment]
Tried it just now and it's no difference. I think the JS sets the "data-checked" before wire:click and it shows the checked state.

I also tried with x-on:click and no luck. Problem is in the Selectable mixin, probably.

**Comment by DinoVincent** [comment]
I've had so many issues with checkboxes. One way to possibly tackle this is by converting the cards/checkboxes to buttons or clickable divs without a checkbox element. And faking a checkbox with styling. Even if you manually change the dom with JS the visual representation of the checkbox can differ from the actual HTML or you get a jitter that changes the checkbox quickly back. 

Fake the checkbox,
